### PR TITLE
Allow JSON upload

### PR DESCRIPTION
**Pull request by MischaU8**
Allow upload of (most) JSON files.

### CODE REVIEW

File: src/components/Upload.tsx [hunk 1]
```diff
@@ -233,6 +234,10 @@ export default function Upload({
         const sheetName = workbook.SheetNames[0]
         const worksheet = workbook.Sheets[sheetName]
         csvContent = XLSX.utils.sheet_to_csv(worksheet)
+      } else if (file.type.includes("json")) {
+        const jsonContent = JSON.parse(e.target?.result as string)
```
Comment (ramonverse) [inline src/components/Upload.tsx:238]: Hi @MischaU8 , your assertion is blocking the app from gracefully failing. Please fix this thanks!

**Comment by ramonverse** [comment]
@MischaU8  If you upload a malformed json the app will forever be loading with no feedback to the user. If this is fixed I'll be happy to merge thanks!
example:
[malformed.json](https://github.com/pretzelai/pretzelai/files/14736271/malformed.json)

**Comment by MischaU8** [comment]
@ramonverse I have added basic error handling and feedback to the user.

Within Pretzel there doesn't seem to be a consistent way to communicate errors to the user, between `console.log`, `console.error`, `alert` and using an `error` state in the component. I chose the latter, but please suggest if you prefer one of the other approaches.

**Comment by ramonverse** [comment]
@MischaU8 looks good for now, we'll add a consistent error handling soon. Merged thanks!

**Pull request merged**